### PR TITLE
archive invalid events and drop existing invalid postgres triggers

### PR DESCRIPTION
**Pull request by tirumaraiselvan**
This is the backport of https://github.com/hasura/graphql-engine/pull/6179 to master

### CODE REVIEW

**Comment by codingkarthik** [comment]
@tirumaraiselvan LGTM, but is there any way to test this?

**Review by codingkarthik — APPROVED** [review]

**Comment by tirumaraiselvan** [comment]
This is the way I tested this:

1. create table test(id serial primary key, name text);
2. Create 2 event triggers:

```
{ "type": "replace_metadata",
  "args":
{
  "version": 2,
  "tables": [
    {
      "table": {
        "schema": "public",
        "name": "test"
      },
      "event_triggers": [
        {
          "name": "test_trigger",
          "definition": {
            "enable_manual": false,
            "insert": {
              "columns": "*"
            }
          },
          "retry_conf": {
            "num_retries": 0,
            "interval_sec": 10,
            "timeout_sec": 60
          },
          "webhook": "http://httpbin.org/post"
        },
        {
          "name": "test_trigger2",
          "definition": {
            "enable_manual": false,
            "insert": {
              "columns": "*"
            }
          },
          "retry_conf": {
            "num_retries": 0,
            "interval_sec": 10,
            "timeout_sec": 60
          },
          "webhook": "http://httpbin.org/post"
        }
      ]
    }
  ]
}
}
```
3. Drop one of the event triggers:

```
{ "type": "replace_metadata",
  "args":
{
  "version": 2,
  "tables": [
    {
      "table": {
        "schema": "public",
        "name": "test"
      },
      "event_triggers":[
        {
          "name": "test_trigger2",
          "definition": {
            "enable_manual": false,
            "insert": {
              "columns": "*"
            }
          },
          "retry_conf": {
            "num_retries": 0,
            "interval_sec": 10,
            "timeout_sec": 60
          },
          "webhook": "http://httpbin.org/post"
        }
      ]
    }
  ]
}
}
```
4. insert into test, 2 events will be created for each event trigger. One of them will throw errors because of https://github.com/hasura/graphql-engine/issues/5461
5. Upgrade with the fix in this PR
6. Notice that the buggy event is now archived

File: server/src-rsr/migrations/39_to_40.sql [hunk 1]
```diff
@@ -24,3 +24,30 @@ $$;
 CREATE TRIGGER event_trigger_table_name_update_trigger
 AFTER UPDATE ON hdb_catalog.hdb_table
 FOR EACH ROW EXECUTE PROCEDURE hdb_catalog.event_trigger_table_name_update();
+
+
+-- this clears pre-existing invalid triggers
```
Comment (jberryman) [inline server/src-rsr/migrations/39_to_40.sql:29]:
```suggestion
-- this clears pre-existing invalid triggers
-- See https://github.com/hasura/graphql-engine/pull/6137
```

**Review by jberryman — APPROVED** [review]
Just that comment for helpful context

**Pull request closed without merging**